activesupport/lib/active_support/cache.rb

```diff
@@ -802,11 +810,11 @@ class Entry # :nodoc:
 
       # Creates a new cache entry for the specified value. Options supported are
       # +:compress+, +:compress_threshold+, +:version+ and +:expires_in+.
-      def initialize(value, compress: true, compress_threshold: DEFAULT_COMPRESS_LIMIT, version: nil, expires_in: nil, **)
+      def initialize(value, compress: true, compress_threshold: DEFAULT_COMPRESS_LIMIT, version: nil, expires_in: nil, expires_at: nil, **)
         @value      = value
         @version    = version
-        @created_at = Time.now.to_f
-        @expires_in = expires_in && expires_in.to_f
+        @created_at = 0.0
```

Comment:
Is this just for backwards-compatibility with serialized data in the cache?

Reply:
Yes, since `Entry` is currently serialized with `Marshal`, we need to preserve the variable names and types.

We could break the format if we wanted, that was done in the past, but I'd rather avoid it. In this case I don't think it's worth it.